Scripts/Python/grsnWallPython.py

```diff
@@ -832,72 +840,93 @@ def SetPanelMode(self,team,disableAll=False,onlNorthLights=False):
                         CountLights[team].value[i].runAttachedResponder(kRedOff)
                 self.oldCount[team] = ageSDL["NumBlockers"][0]
             elif(state >= kSetBlocker):
-                for i in range(0,ageSDL["NumBlockers"][0]):
-                    if(i < self.GetNumBlockerSet(team)):
-                        CountLights[team].value[i].runAttachedResponder(kTeamLightsOn)
-                    else:
-                        CountLights[team].value[i].runAttachedResponder(kRedOn)
+                redLightCount = self.GetNumBlockerSet(team)
+                if redLightCount == 0:
+                    CountLights[team].value[redLightCount].runAttachedResponder(kRedOn)
+                elif 1 <= redLightCount < ageSDL["NumBlockers"][0]:
+                    CountLights[team].value[redLightCount-1].runAttachedResponder(kTeamLightsOn)
+                    CountLights[team].value[redLightCount].runAttachedResponder(kRedOn)
+                else:
+                    CountLights[team].value[redLightCount-1].runAttachedResponder(kTeamLightsOn)
             return
 
         PtDebugPrint("grsnWallPython::SetPanelMode: %s - %d" % (team, state))
-        if(state == kStandby or state == kEnd or disableAll):
-            for i in range(0,20):
-                CountLights[team].value[i].runAttachedResponder(kRedOff)
-            goBtnObject[team].value.runAttachedResponder(kDim)
+        if(disableAll):
+            for i in range(0,171):
+                Panel[team].value[i].physics.disable()
+            #goBtnObject[team].value.runAttachedResponder(kDim)
             goButton[team].disable()
             upButton[team].disable()
             dnButton[team].disable()
             readyButton[team].disable()
             fiveBtn[team].disable()
             tenBtn[team].disable()
             fifteenBtn[team].disable()
-        elif(state == kSit):
-            goBtnObject[team].value.runAttachedResponder(kBright)
-            goButton[team].enable()
+        elif(state == kStandby or state == kEnd):
+            for i in range(0,20):
+                CountLights[team].value[i].runAttachedResponder(kRedOff)
+            for i in range(0,171):
+                Panel[team].value[i].physics.disable()
```

Comment:
Are these loops iterating all of the lights and panel objects?

Reply:
Yes, they turn on/off the lights or physics for the panels